### DISCUSSION TITLE
Missing set{Data,Markdown,Html}TemplateEngine configuration methods

### DISCUSSION THREAD

**Discussion by valtlai** [11ty/eleventy#4073] [Enhancement Queue]
**Is your feature request related to a problem? Please describe.**
There’s no corresponding methods for the `dataTemplateEngine`, `markdownTemplateEngine` and `htmlTemplateEngine` [configuration properties](https://www.11ty.dev/docs/config/#default-template-engine-for-global-data-files).

**Describe the solution you'd like**
Adding the `setDataTemplateEngine`, `setMarkdownTemplateEngine` and `setHtmlTemplateEngine` configuration methods.

**Describe alternatives you've considered**
Using the existing properties.

**Additional context**
I’d like to only use methods for consistency.

Also see #1503 for `dir`.

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by zachleat** [comment]
I tried to reopen https://github.com/11ty/eleventy/issues/1502 after seeing this was added in v4.0.0 canaries but unfortunately we’ve moved to #4079 to track this in the 4.0 milestone
